src/Error/BaseErrorHandler.php

```diff
@@ -297,6 +298,11 @@ protected function _getMessage(Exception $exception)
             if ($request) {
                 $message .= "\nRequest URL: " . $request->here();
             }
+
+            $request = new Request();
```

Comment:
Thats already done at https://github.com/cakephp/cakephp/pull/8089/files#diff-543eecb5875a400afd68469e05fc933fR297
I would just put the code after the line there, no additional check required.